drivers/kscan/kscan_npcx.c

```diff
@@ -0,0 +1,230 @@
+		mask = ~(BIT(CONFIG_KSCAN_MATRIX_MAX_COLUMNS) - 1);
+	}
+	/* Drive one line to low for determining which key's state changed. */
+	else {
+		mask = ~BIT(col);
+	}
+
+	LOG_DBG("%x", mask);
+
+	/* Set KBSOUT */
+	inst->KBSOUT0 = (mask & 0xFFFF);
+	inst->KBSOUT1 = ((mask >> 16) & 0x03);
+
+	return 0;
+}
+
+static int kscan_npcx_read_matrix_row(const struct device *dev, int *row)
+{
+	const struct kscan_npcx_config *config = dev->config;
+	struct kbs_reg *const inst = config->base;
+	int val;
+
+	val = inst->KBSIN;
+
+	/* 1 means key pressed, otherwise means key released. */
+	*row = (~val & NPCX_KB_ROW_MASK);
+
+	return 0;
+}
+
+static int kscan_npcx_resume_matrix_detection(const struct device *dev, bool resume)
+{
+	const struct kscan_npcx_config *const config = dev->config;
+
+	if (resume) {
+		irq_enable(config->irq);
+	} else {
+		irq_disable(config->irq);
+	}
+
+	return 0;
+}
+
+static int kscan_npcx_init(const struct device *dev)
+{
+	const struct device *clk_dev = DEVICE_DT_GET(NPCX_CLK_CTRL_NODE);
+	const struct kscan_npcx_config *const config = dev->config;
+	struct kbs_reg *const inst = config->base;
+	int ret;
+
+	if (!device_is_ready(clk_dev)) {
+		LOG_ERR("%s device not ready", clk_dev->name);
+		return -ENODEV;
+	}
+
+	/* Turn on KBSCAN controller device clock */
+	ret = clock_control_on(clk_dev, (clock_control_subsys_t *)&config->clk_cfg);
+	if (ret < 0) {
+		LOG_ERR("Turn on KBSCAN clock fail %d", ret);
+	}
+
+	/* Pull-up KBSIN0-7 internally */
+	inst->KBSINPU = 0xFF;
+
+	/*
+	 * Keyboard Scan Control Register
+	 *
+	 * [6:7] - KBHDRV KBSOUTn signals output buffers are open-drain.
+	 * [3] - KBSINC   Auto-increment of Buffer Data register is disabled
+	 * [2] - KBSIEN   Interrupt of Auto-Scan is disabled
+	 * [1] - KBSMODE  Key detection mechanism is implemented by firmware
+	 * [0] - START    Write 0 to this field is not affected
+	 */
+	inst->KBSCTL = 0x00;
+
+	/*
+	 * Select quasi-bidirectional buffers for KSO pins. It reduces the
+	 * low-to-high transition time. This feature only supports in npcx7.
+	 */
+	if (IS_ENABLED(CONFIG_KSCAN_NPCX_KSO_HIGH_DRIVE)) {
+		SET_FIELD(inst->KBSCTL, NPCX_KBSCTL_KBHDRV_FIELD, 0x01);
+	}
+
+	/* Drive all column lines to low for detection any key press */
+	kscan_npcx_drive_matrix_column(dev, KEYBOARD_COLUMN_DRIVE_NONE);
+
+	/* Configure wake-up input and callback for keyboard input signal */
+	for (int i = 0; i < ARRAY_SIZE(ksi_callback); i++)
+		kscan_npcx_init_ksi_wui_callback(dev, &ksi_callback[i], &config->wui_maps[i],
+						 kscan_npcx_ksi_isr);
+
+	/* Configure pin-mux for kscan device */
+	npcx_pinctrl_mux_configure(config->alts_list, config->alts_size, 1);
+
+	return 0;
+}
+
+static const struct kscan_matrix_driver_api kscan_matrix_npcx_driver_api = {
+	.config_matrix = kscan_npcx_configure_matrix,
+	.drive_matrix_column = kscan_npcx_drive_matrix_column,
+	.read_matrix_row = kscan_npcx_read_matrix_row,
+	.resume_matrix_detection = kscan_npcx_resume_matrix_detection,
```

Comment:
Update the naming of the local functions to match kscan_matrix_driver_api.
```
	.matrix_config = kscan_matrix_npcx_config,
	.matrix_drive_column = kscan_matrix_npcx_drive_column,
	.matrix_read_row = kscan_matrix_npcx_read_row,
	.matrix_resume_detection = kscan_matrix_npcx_resume_detection,```

Reply:
Done.